builders/interoperability/mrl.md

```diff
@@ -0,0 +1,152 @@
+
+In the future, the xTokens pallet will be updated, allowing for your native gas currency to be used as a fee token instead. Parachains that use a different pallet would need to implement their own solution to transfer reserve and non-reserve assets in a single message.  
+
+![Transfering wormhole MRL out](/images/builders/interoperability/mrl/mrl-2.png)
+
+In the diagram above, the user is sending MRL tokens from a parachain back into Wormhole.
+
+1. The user sends two XCM messages in one transaction using the utility.batchAll extrinsic:
+2. The first XCM message sends xcGLMR & the ERC-20 to the user’s multilocation derivative account with `xTokens.transferMultiassets`, using xcGLMR as the fee token. The multilocation-derivative account is a keyless account on Moonbeam that an account on another parachain has control of via XCM
+3. The second XCM message sends a remote EVM call to the batch precompile via polkadotXcm.send extrinsic
+    - First, it approves the WormholeTokenBridge smart contract to move the ERC-20 sent over in the second step 
+    - Second, it sends tokens across chains with the WormholeTokenBridge’s transferTokens function
+4. The guardian network will pick up on the Wormhole transaction and sign it 
+5. A wormhole relayer will relay the tokens to the destination chain & destination account
+
+There are many components to this batch transaction. To construct a transaction, you can write the following code using `polkadot.js` and `ethers` packages.  
+
+```javascript
+import { ApiPromise, WsProvider } from '@polkadot/api';
+import Keyring from '@polkadot/keyring';
+import { BN } from "@polkadot/util";
+import { ethers, providers } from 'ethers';
+```
+
+The batch transaction's first responsibility is to send xcGLMR and the ERC-20 to the multilocation derivative account. It is recommended to calculate the multilocation derivative account for the user, which can be calculated off-chain using the script from the [xcm-tools repository](https://github.com/PureStake/xcm-tools){target=_blank}. Alternatively, the [XCM-Utils precompile](/builders/pallets-precompiles/precompiles/xcm-utils) can also be used.  
+
+xcGLMR must be used as a fee token in `xTokens.transferMultiassets` as opposed to the ERC-20 token because of a restriction with how the xTokens pallet currently works.  
+
+??? code "xTokens.transferMultiassets Extrinsic Construction" 
+    ```js
+    --8<-- 'code/mrl/transfer-multiassets.js'
+    ```
+
+To generate the second end of the transaction (remote EVM call), you need both the EVM transaction and the XCM instructions that executes said EVM transaction. The EVM transaction can be constructed as a transaction that interacts with the [batch precompile](/builders/pallets-precompiles/precompiles/batch){target=_blank} so that two transactions can happen in one. This is helpful because this EVM transaction has to both approve the xLabs relayer to relay the ERC-20 token as well as relay action itself.  
```

Comment:
can this be any relayer?

Reply:
Yes, any relayer. Relayers are trustless